### DISCUSSION TITLE
Instances autoscaling

### DISCUSSION THREAD

**Discussion by vfdev-5** [ComputeCanada/magic_castle#185] [Q&A]
Hi, I wonder if we can setup autoscaling option for instances:
- https://github.com/ComputeCanada/magic_castle/tree/main/docs#47-instances

For example, we would like to have 0 gpu node and that could be created by the cloud provider up to a max count:
```
instances = {
  ...
  gpu-node = { type = "gpu2.large", count = 0, max_count=4, tags = [...] },
}
```

**Comment by ocaisa** [comment]
Maybe take a look through #170 to see the status of the ideas for this. The core issues are whether this can be done in a universal way, and what size is the security hole left sitting on the cluster to make this work.
